pkg/reactor/reactor.go

```diff
@@ -165,55 +169,189 @@ func printResolved(ctx context.Context, manifest *api.Documentation, writer io.W
 // - NodeSelectors
 // The resulting model is the actual flight plan for replicating resources.
 func (r *Reactor) resolveStructure(ctx context.Context, nodes []*api.Node, globalLinksConfig *api.Links) error {
-	var handler resourcehandlers.ResourceHandler
 	for _, node := range nodes {
 		node.SetParentsDownwards()
 		if len(node.Source) > 0 {
-			if handler = r.ResourceHandlers.Get(node.Source); handler == nil {
-				return fmt.Errorf("No suitable handler registered for URL %s", node.Source)
-			}
-			if len(node.Name) == 0 {
-				node.Name = "$name"
+			nodeName, err := r.resolveNodeName(ctx, node)
+			if err != nil {
+				return err
 			}
-			name, ext := handler.ResourceName(node.Source)
-			id := uuid.New().String()
-			node.Name = strings.ReplaceAll(node.Name, "$name", name)
-			node.Name = strings.ReplaceAll(node.Name, "$uuid", id)
-			node.Name = strings.ReplaceAll(node.Name, "$ext", fmt.Sprintf(".%s", ext))
+			node.Name = nodeName
 		}
 		if node.NodeSelector != nil {
-			if handler = r.ResourceHandlers.Get(node.NodeSelector.Path); handler == nil {
-				return fmt.Errorf("No suitable handler registered for path %s", node.NodeSelector.Path)
-			}
-			if err := handler.ResolveNodeSelector(ctx, node, node.NodeSelector.ExcludePaths, node.NodeSelector.FrontMatter, node.NodeSelector.ExcludeFrontMatter, node.NodeSelector.Depth); err != nil {
+			newNode, err := r.resolveNodeSelector(ctx, node, globalLinksConfig)
+			if err != nil {
 				return err
 			}
-			// remove node selectors after resolution
+			node.Nodes = append(node.Nodes, newNode.Nodes...)
```

Comment:
Optionally, this node may already contain nodes with the same names. What's expected to happen is nodes with identical names to get merged. Overlapping property conflicts are resolved in favor of the explicitly defined node (and _not_ the one coming from nodeSelector).  That's a way to not only mashup, but also _override_ or _inject_ properties into nodes resolved form a `nodesSelector`. 

In this case however I think nodes with the same name will be duplicated in the `nodes` slice instead.

Consider the following:
```
nodesSelector:
  path: <something-that-will-resolve-to-node-with-name-A>
nodes:
- name A
  properties:
     a: b
```
and the node that will be resolved from the nodeSelector as child of `nodes`:
```
- name A
  properties:
     a: c
```
The expected behavior is that we end up with this:
```
nodesSelector:
  path: <something-that-will-resolve-to-node-with-name-A>
nodes:
- name A
  properties:
     a: b <- overwrites a:c
```
but instead, it seems we will end up with this:
```
nodesSelector:
  path: <something-that-will-resolve-to-node-with-name-A>
nodes:
- name A
  properties:
     a: b
- name A
  properties:
     a: c
```

Please, describe this as a next step enhancement request.